base_layer/core/src/transactions/key_manager/inner.rs

```diff
@@ -300,20 +278,144 @@ where TBackend: KeyManagerBackend<PublicKey> + 'static
         }
     }
 
-    fn get_domain_hasher(
-        label: &str,
-    ) -> Result<DomainSeparatedHasher<Blake2b<U64>, KeyManagerTransactionsHashDomain>, KeyManagerServiceError> {
-        let tx_label = label.parse::<TransactionKeyManagerLabel>().map_err(|e| {
-            KeyManagerServiceError::UnknownError(format!("Could not retrieve label for derived key: {}", e))
-        })?;
-        match tx_label {
-            TransactionKeyManagerLabel::ScriptKey => Ok(DomainSeparatedHasher::<
-                Blake2b<U64>,
-                KeyManagerTransactionsHashDomain,
-            >::new_with_label("script key")),
+    pub(crate) async fn get_private_key(&self, key_id: &TariKeyId) -> Result<PrivateKey, KeyManagerServiceError> {
+        match key_id {
+            KeyId::Managed { branch, index } => {
+                // ledger has special rules here
+                if let WalletType::Ledger(wallet) = &self.wallet_type {
```

Comment:
nit: It might be more effective here to use a `match` statement so that if a future `WalletType` is added we know to change this code